wiring/inc/spark_wiring_cloud_event.h

```diff
@@ -0,0 +1,477 @@
+        return this->data(data.data(), data.size());
+    }
+
+    /**
+     * Set the event data.
+     *
+     * @param data Event data.
+     * @param type Content type.
+     * @return This event instance.
+     */
+    CloudEvent& data(const Buffer& data, ContentType type) {
+        return this->data(data.data(), data.size(), type);
+    }
+
+    /**
+     * Set the event data.
+     *
+     * @param data Event data.
+     * @return This event instance.
+     */
+    CloudEvent& data(const Variant& data);
+
+    /**
+     * Get the event data.
+     *
+     * This method returns a copy of the event data in a dynamically allocated buffer.
+     *
+     * @return Event data.
+     */
+    Buffer data() const;
+
+    /**
+     * Get the event data.
+     *
+     * @return Event data.
+     */
+    String dataAsString() const;
+
+    /**
+     * Get the event data.
+     *
+     * @return Event data.
+     */
+    Variant dataAsVariant() /* FIXME: const */; // TODO: Rename?
+
+    /**
+     * Load the event data from a file.
+     *
+     * @param path File path.
+     * @return This event instance.
+     */
+    CloudEvent& loadData(const char* path);
+
+    /**
+     * Save the event data to a file.
+     *
+     * @param path File path.
+     * @return 0 on success, otherwise an error code defined by `Error::Type`.
+     */
+    int saveData(const char* path);
+
+    /**
+     * Check if the event has payload data.
+     *
+     * @return `true` if the event has payload data, otherwise `false`.
+     */
+    bool hasData() const {
+        return size() > 0;
+    }
+
+    /**
+     * Set the maximum size of event data that can be store on the heap.
+     *
+     * The data exceeding the specified size will be stored in a temporary file.
+     *
+     * This method has no effect if the event already contains any data.
+     *
+     * The default value is 1024 bytes.
+     *
+     * @param size Data size.
+     * @return This event instance.
+     */
+    CloudEvent& maxDataSizeInRam(size_t size);
+
+    /**
+     * Get the maximum size of event data that can be store on the heap.
+     *
+     * @return Data size.
+     */
+    size_t maxDataSizeInRam() const;
+
+    /**
+     * Set a callback to be invoked when the status of the event changes.
+     *
+     * @param callback Callback.
+     * @return This event instance.
+     */
+    CloudEvent& onStatusChange(std::function<OnStatusChange> callback);
+
+    /**
+     * Get the status of the event.
+     *
+     * @return Event status.
+     */
+    Status status() const;
+
+    /**
+     * Check if the event is valid.
+     *
+     * @return `true` if the event is valid, otherwise `false`.
+     */
+    bool valid() const {
+        return status() != Status::INVALID;
+    }
+
+    /**
+     * Check if the event is being sent to the Cloud.
+     *
+     * @return `true` if the event is being sent to the Cloud, otherwise `false`.
+     */
+    bool sending() {
```

Comment:
There's no shorthand to see if a CloudEvent is in NEW state. Can you add a `new()` method?

Reply:
`new` is a reserved keyword so it will be `isNew()`. For consistency, I'll need to rename all the other shorthand methods to have an `is` prefix which I omitted for brevity.

Reply:
Or rename the initial state PENDING

Reply:
I added `isNew` and renamed the other methods to `isSending`, `isOk`, etc. `PENDING` is a good but may be confusing for events in progress. Also, incoming events are received in that state too.